### PR TITLE
suggestion: incremental hash calculation

### PR DESCRIPTION
**Pull request by folkertvanheusden**
Hi,

I've tried to implement incremental hash calculation (partially).

### CODE REVIEW

File: tests/PositionTests.cpp [hunk 1]
```diff
@@ -35,12 +35,22 @@ TEST_CASE("Null Move Test", "[Position]") {
...
     REQUIRE(pos.hash() == old_hash);
+
+    pos.make_move({B1, C3});
+    REQUIRE(pos.hash() != old_hash);
+    pos.make_move({B8, C6});
+    REQUIRE(pos.hash() != old_hash);
+    pos.make_move({C3, B1});
+    REQUIRE(pos.hash() != old_hash);
+    pos.make_move({C6, B8});
+    REQUIRE(pos.hash() == old_hash);
```
Comment (Mk-Chan) [inline tests/PositionTests.cpp:63]: Add a new test case which checks whether the make_move (in each of the cases you added incremental hash) causes new hash to be equal to calculate_hash() from scratch. This proves it's equivalent to the old functionality.

Comment (folkertvanheusden) [reply tests/PositionTests.cpp:63]: Calculate_hash is a private member. Can I make it public?

Comment (Mk-Chan) [reply tests/PositionTests.cpp:63]: Yeah I guess it's fine

Comment (folkertvanheusden) [reply tests/PositionTests.cpp:63]: Note that even with this partial hash-update, the perft-batch went from 01:55 to 01:40 on my laptop.

File: tests/PositionTests.cpp [hunk 2]
```diff
@@ -35,22 +35,25 @@ TEST_CASE("Null Move Test", "[Position]") {
...
     REQUIRE(pos.hash() != old_hash);
+    REQUIRE(pos.hash() == pos.calculate_hash());
     pos.unmake_move();
     REQUIRE(pos.hash() == old_hash);
 
     pos.make_move({B1, C3});
     REQUIRE(pos.hash() != old_hash);
+    REQUIRE(pos.hash() == pos.calculate_hash());
     pos.make_move({B8, C6});
-    REQUIRE(pos.hash() != old_hash);
```
Comment (Mk-Chan) [inline tests/PositionTests.cpp]: Don't remove this, the test is still correct

Comment (folkertvanheusden) [reply tests/PositionTests.cpp]: done

**Review by Mk-Chan — APPROVED** [review]

**Pull request merged**